tycho-its/src/test/java/org/eclipse/tycho/test/baseline/BaselineMojoTest.java

```diff
@@ -55,17 +75,43 @@ public void testUnchangedApi() throws Exception {
 	 * @throws Exception
 	 */
 	@Test
-	public void testAddMethod() throws Exception {
+	public void testAddMethod() throws Throwable {
+
 		// test adding a default method to the "public" interface
-		Verifier verifier;
-		verifier = buildBaselineProject("add-method", true);
-		verifyBaselineProblem(verifier, "ADDED", "METHOD", "concat(java.lang.String,java.lang.String)", "1.0.0",
-				"1.1.0");
+		Verifier verifier1 = buildBaselineProject(true, r -> {
+			Path aClass = r.resolve(Path.of("src", "my", "api", "bundle", "MyApiInterface.java"));
+			replaceStringInFileWithPlatformsLineSeparator(aClass, """
```

Comment:
This here indicates one can use the java AST outside of eclipse:
https://stackoverflow.com/a/20918431

this would avoid using string replace operations at all.

Reply:
Looks promising, I started to do that, but running out of time. Will complete it later today.
Then the string replaces should not be necessary at all. 👍🏽

Reply:
I think we are not in hurry but this can greatly simplify test creation and make it more "semantic", so I think its worth to be patient :-)

Reply:
Is completed now. ;)